### ISSUE TITLE
The custom layer is under the building layer

### ISSUE THREAD

**Issue by limzgiser** [peterqliu/threebox#80]
Can someone help me,Why can't I move the layer above the building layer?

**Comment by zearg** [comment]
I've same problem with mapbox version > 0.54

Have you found a fix ?

**Comment by joedjc** [comment]
Same problem here...doesn't seem to be an issue with custom-layers generally

**Comment by zearg** [comment]
There is more information in this issue: https://github.com/mapbox/mapbox-gl-js/issues/7395

peterqliu commented on 16 Nov 2018
> [...] The custom layer draws within the context of the 2D layer stack, regardless of whether the layers themselves are 3D. For proper depth testing, you would put all 3D elements into the same custom layer

**Comment by zearg** [comment]
You can add 

`gl.clear(gl.DEPTH_BUFFER_BIT);`

in render() of layer in first position (before all actions in this function).
But three objects will be always above buildings.

After tests, CameraSync is the thing that 3D depth calcul isn't working.
Maybe `this.camera.matrixWorld.copy()` or translation of the world group... i don't know.

I think i will create a world group, positionned at 0,0,0 and translate camera instead of translating world group.
Difficult is to convert world coords to and from lat/lng.

**Comment by sys2007** [comment]
map.moveLayer('id1','id2')

**Comment by adelchamas96** [comment]
yes  i got same issue  after using `gl.clear(gl.DEPTH_BUFFER_BIT);` , also i tried to change ordering nothing work correctly. here a screenshot:
![Screenshot (18)](https://user-images.githubusercontent.com/60884385/80683537-08df4480-8acd-11ea-9822-1a0227bbedc8.png)

anyone solved this issue?

**Comment by jscastro76** [comment]
Hi @adelchamas96 , @limzgiser , @joedjc I I have solved this problem at my [fork](https://github.com/jscastro76/threebox/) fine tuning the CameraSync object. 
![image](https://user-images.githubusercontent.com/38470852/85045206-f27d6d00-b18e-11ea-9a2d-3b9c4a850b81.png)
![image](https://user-images.githubusercontent.com/38470852/85045318-18a30d00-b18f-11ea-91a8-d5a59d6719de.png)
